File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.js
function solution(my_string, s, e) {
    let arr = my_string.split(""); // 문자열 → 문자 배열
    const reversedArr = arr.slice(s, e + 1).reverse(); // s~e 구간 잘라서 뒤집기
    arr.splice(s, e - s + 1, ...reversedArr); // 배열 요소를 펼쳐서 교체
    return arr.join(""); // 다시 문자열로 변환
}

File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.java
class Solution {
    public String solution(String my_string, int s, int e) {
        // 1. 문자열을 문자 배열로 변환
        char[] arr = my_string.toCharArray();
        
        // 2. s~e 구간을 뒤집기
        while (s < e) {
            char temp = arr[s];
            arr[s] = arr[e];
            arr[e] = temp;
            s++;
            e--;
        }
        
        // 3. 문자 배열을 다시 문자열로 변환
        return new String(arr);
    }
}
